From 56000ac455cf5ddc582f59029a7ca1dd3d5e1095 Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@nxp.com>
Date: Mon, 29 Jan 2018 16:57:18 +0800
Subject: [PATCH 306/448] vfio fsl-mc: Enable VFIO read/writes for DPMCP objects

MC commands can come over DPMCP objects as well, so we enabled
read/writes on DPMCP objects as well.

Signed-off-by: Zhao Qiang <qiang.zhao@nxp.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc.c b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
index 9dc32d2..1ec2b31 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
@@ -289,8 +289,9 @@ static ssize_t vfio_fsl_mc_read(void *device_data, char __user *buf,
 	uint64_t data[8];
 	int i;
 
-	/* Read ioctl supported only for DPRC device */
-	if (strcmp(vdev->mc_dev->obj_desc.type, "dprc"))
+	/* Read ioctl supported only for DPRC and DPMCP device */
+	if (strcmp(vdev->mc_dev->obj_desc.type, "dprc") &&
+	    strcmp(vdev->mc_dev->obj_desc.type, "dpmcp"))
 		return -EINVAL;
 
 	if (index >= vdev->num_regions)
@@ -391,8 +392,9 @@ static ssize_t vfio_fsl_mc_write(void *device_data, const char __user *buf,
 	uint64_t data[8];
 	int ret;
 
-	/* Write ioctl supported only for DPRC device */
-	if (strcmp(vdev->mc_dev->obj_desc.type, "dprc"))
+	/* Write ioctl supported only for DPRC and DPMCP device */
+	if (strcmp(vdev->mc_dev->obj_desc.type, "dprc") &&
+	    strcmp(vdev->mc_dev->obj_desc.type, "dpmcp"))
 		return -EINVAL;
 
 	if (index >= vdev->num_regions)
-- 
1.7.1

